component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java

```diff
@@ -95,6 +95,14 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
             throws AuthenticationFailedException {
         String username = null;
         AuthenticatedUser authenticatedUser;
+        try {
+            String tenantDomain = context.getTenantDomain();
+            if (!tenantDomain.equals(TOTPAuthenticatorConstants.SUPER_TENANT_DOMAIN)) {
+                TOTPUtil.loadXMLFromRegistry(context, tenantDomain);
+            }
+        } catch (TOTPException e) {
```

Comment:
I already tested this usecase. loadXMLFromRegistry(context, tenantDomain) method will handle if there is no entry in registry for tenant.